account_invoice_triple_discount/models/account_move_line.py

```diff
@@ -33,28 +33,20 @@ class AccountMoveLine(models.Model):
     @api.depends("discount1", "discount2", "discount3")
     def _compute_discount(self):
         for line in self:
-            line.discount = line._get_aggregated_discount_from_values(
-                {
-                    fname: line[fname]
-                    for fname in line._get_multiple_discount_field_names()
-                }
+            line.discount = line._get_aggregated_multiple_discounts(
+                [line[x] for x in ["discount1", "discount2", "discount3"]]
             )
 
-    def _get_aggregated_discount_from_values(self, values):
-        discount_fnames = self._get_multiple_discount_field_names()
-        discounts = []
-        for discount_fname in discount_fnames:
-            discounts.append(values.get(discount_fname) or 0.0)
-        return self._get_aggregated_multiple_discounts(discounts)
-
     def _get_aggregated_multiple_discounts(self, discounts):
+        """
+        Returns the aggregate discount corresponding to any number of discounts.
+        For exemple, if discounts is [11.0, 22.0, 33.0]
+        It will return 46.5114
+        """
         discount_values = []
         for discount in discounts:
             discount_values.append(1 - (discount or 0.0) / 100.0)
         aggregated_discount = (
             1 - functools.reduce((lambda x, y: x * y), discount_values)
         ) * 100
         return aggregated_discount
-
-    def _get_multiple_discount_field_names(self):
```

Comment:
I would keep this function if we can use its result in different places

Reply:
indeed ! restored !